Updated to Python3 / Gtk3 for >= Ubuntu 20.04 support

cdberkstresser opened the pull request:
This is a fairly intense update from Python2 / Gtk2 to Python3 / Gtk 3.  The Glade xml file is completely gtk3 with no deprecated widgets and should allow easier maintenance into the future without the backward dependencies that are being phased out like python-support and python-gtk2.

r-barnes committed:
Upgrade exception handling to Python3

Co-Authored-By: Matthew Harm Bekkema <id@mbekkema.name>

nikolaik committed:
Get running on macos

Remove is_old_gtk and it's usage
Remove osx_window_focus_workaround since it's probably not needed
anymore with gtk3

cdberkstresser committed:
Initial buggy port to python3

cdberkstresser committed:
Updating some glade stuff

cdberkstresser committed:
changing out table and renaming labels

cdberkstresser committed:
Finished changing all tables to grids

cdberkstresser committed:
converting hboxes and vboxes

cdberkstresser committed:
finished converting deprecated widgets

cdberkstresser committed:
Fixing resize errors

cdberkstresser committed:
Fixing crash on config open and left aligning project link

cdberkstresser committed:
Fixing icons for program

cdberkstresser committed:
Fixing working directory errors for loading icons

cdberkstresser committed:
Fixing multiple cores and work units

cdberkstresser committed:
adding try/catch for version

ffissore committed:
Wrapping labels of slot dialog

ffissore committed:
Converting slot option `value` to string, in order to avoid `ValueError`

cdberkstresser committed:
Merge branch 'ffissore-master'
Merging in changes from ffissore

Wrapping labels of slot dialog
Converting slot option `value` to string, in order to avoid `ValueError`

ffissore commented on fah/Client.py:
```diff
@@ -316,7 +316,7 @@ def process_error(self, app, data):
             app.error(msg)
 
-        else: print('ERROR: %s' % msg)
+        else: print(('ERROR: %s' % msg))
```
I think the double parenthesis is not necessary

cdberkstresser replied:
agreed

ffissore committed:
Assuming first entry of comboboxes is the default one, select it if no value is provided

ffissore committed:
Removing unnecessary double parentheses when calling `print`

cdberkstresser committed:
linting round 0

ffissore commented:
I'll jot down here two things I've found. Unless you fix them in the meanwhile, I'll fix them asap
- when saving configuration with a changed username, a `b` is prepended. this is due to [unicode_escape](https://github.com/cdberkstresser/fah-control/blob/6cf51bf34c4071494629eab63ace763efb3000a2/fah/Client.py#L201). Since `string_escape` is apparently gone, a simple `value.replace('\'', '\\\'')` should do
- when saving configuration, the cause is lost (it becomes an empty string)

cdberkstresser committed:
Adding encoding utf8 to run on windows

cdberkstresser committed:
Using Default Geometry from Glade

cdberkstresser committed:
Bug Fixes 0

unicode_escape replacement
saving cause now
attaching donor, team, and cause to combo box model indices correctly in glade

cdberkstresser commented:
@ffissore I believe those are fixed now :)

cdberkstresser committed:
Fixing theme and rendering combos

ffissore committed:
Fixed donor and team button links

ffissore committed:
Made get_widget_str_value more generic when getting the selected value of a combobox

ffissore committed:
Updated project stats URL

ffissore committed:
Removed a warning about not having specified the gtk version

cdberkstresser committed:
Changing is not -1 literal to != -1

cdberkstresser committed:
Merge branch 'fixes'

Merging ffissore's Fixes request.  Thanks!

ffissore committed:
Fixed "copy log" button

ffissore committed:
Wrapping text of the about dialog

cdberkstresser committed:
Merge pull request #3 from cdberkstresser/fixes

Fixes

cdberkstresser committed:
init

cdberkstresser committed:
utf8 and disable icon

cdberkstresser committed:
Removing about icon from help -> about

cdberkstresser committed:
Fixing constructor for Connection

cdberkstresser committed:
fixing pygobject dependency

braiam committed:
[debian] Create debian packaging information

There's some miscelaneous fixes that needs to be
applied for a full compliance. But this is enough
to create a package.

Full policy:
https://wiki.debian.org/Python/LibraryStyleGuide

cdberkstresser commented:
I think we should probably pause for a review here to see how much to throw into this PR.

ffissore committed:
Sorting queue entries

ffissore committed:
Sorting queue entries

ffissore committed:
Show progress percentage on progress bar

ffissore committed:
Show progress percentage on progress bar

ffissore committed:
Restored original font size on toolbar buttons

ffissore committed:
Restored original font size on toolbar buttons

cdberkstresser commented:
@jcoffland Any thoughts on what you would like for this pull request or if you would like us to continue?

cdberkstresser committed:
Merge pull request #10 from cdberkstresser/fixes

Merging in changes for sorting queue and progresss bar enhancements

ariedov commented:
Can we have a build of that? Just, you know, for testing. 😅

ffissore commented:
@ariedov we started working on https://github.com/cdberkstresser/fah-control/issues/8 which will provide canary builds

bhack commented:
Are you aware of https://github.com/FoldingAtHome/fah-control/pull/56?

ffissore commented:
To be honest, I became aware of the other PRs just yesterday

guystreeter commented:
I've been helping with that other PR. It intends to be "only the necessary changes", so no reformatting or other re-writes. It appears to be working where it has been tested.

cdberkstresser commented:
@bhack  @guystreeter  Hopefully the other PR goes well for you all.  This PR was originally just an attempt to learn python3/gtk3 to get it going for beta testing the new fahclient; however, it does not intend to be "only the necessary changes" and intends rather to be free to explore more avenues of functionality towards beta testing and UI changes so I hope it is different enough not to feel like a duplicated effort.  Just looking to see FAH do well into the future and not trying to step on anyone's work. :)  Kudos for #56 !

cdberkstresser committed:
Merge branch 'master' of https://github.com/braiam/fah-control into braiam-master
merging braiam deb files

cdberkstresser committed:
changes to compat and changelog to version 7.7.0

cdberkstresser committed:
Merge branch 'braiam-master'
Merging with minor changes to braiam deb configuration files

cdberkstresser committed:
fixing expert settings overflow

cdberkstresser committed:
Workflow for packaging debian file

braiam committed:
debian/control: Moved build dependencies and wrap-and-sort

mat8913 committed:
Use bytes literals where appropriate

ffissore committed:
Revert "Removing about icon from help -> about"

This reverts commit ad4a993262ea46a7c4bc8319f67e57bd9b71c55c.

ffissore commented:
@mat8913 @nikolaik @guystreeter @r-barnes @cdberkstresser FYI I'm testing the other PRs and collecting their goodness giving credit to the respective authors

cdberkstresser committed:
Merge https://github.com/braiam/fah-control into braiam-master

cdberkstresser committed:
Merge branch 'braiam-master'
Merging braiam's control changes for building debs

cdberkstresser committed:
Updating workflow to install debian package

cdberkstresser committed:
Reverting package workflow install

cdberkstresser committed:
Update packagedeb.yml

cdberkstresser committed:
reverting workflow.

ffissore committed:
Imports cleanup

nikolaik committed:
Porting changes from #56

guystreeter committed:
Porting changes from #56

ffissore committed:
Tuning UI margins

ffissore committed:
Fixed about dialog icon and orientation

ffissore committed:
Fixed app icon loading mechanism

ffissore committed:
Fixed preferences dialog orientation

cdberkstresser committed:
Fixing Wash U credit

cdberkstresser committed:
Merge pull request #14 from cdberkstresser/goodness-from-other-prs

Goodness from other PRs + Fixes

ffissore commented:
@mat8913 @nikolaik @guystreeter @r-barnes Please take a look at https://github.com/cdberkstresser/fah-control/pull/14: it's made of made-up commits with you set as authors
@cdberkstresser I have **not** included a bunch of commits from @guystreeter related to RPM packaging, since I don't have a fedora to test them: pls consider [cherry-picking them](https://github.com/FoldingAtHome/fah-control/pull/56/commits)

guystreeter commented:
I tested the rpm changes for #56 locally on Fedora 31 and 32, both on my installed system and in a Mock buildroot. I used the srpm from the Mock build to build for Fedora and Centos 8 on COPR, the public Fedora build system. rpm packages built from #56 are available for installation and testing at [COPR](https://copr.fedorainfracloud.org/coprs/streeter/FAHControl/). I have not tested the rpm on Centos 8 because I don't have a test system.
It appears to me that my changes should work fine for this PR. They may need to be worked into whatever build environment FAH uses. Also, I left Copyright, Licensing, and Version as "to-do" items. I am fine with assigning the copyright of the new files to FAH.

informatorius committed:
Don't close FahViewer on FahControl close

cdberkstresser committed:
Merge branch 'issue#1385' of https://github.com/informatorius/fah-control
Fixing viewer close on control exit

ffissore committed:
Fixed self-changing slot/queue selection

ffissore committed:
Workaround for weird log pane scrolling behaviour.
Fixes https://github.com/cdberkstresser/fah-control/issues/9

cdberkstresser committed:
Merge pull request #15 from cdberkstresser/self-selection

Fixed self-changing slot/queue selection

yumetodo commented:
BTW, in my environment, `python setup.py --command-packages=stdeb.command sdist_dsc` doesn't work.

```
sudo sudo apt install python3-all debhelper python3-setuptools devscripts
$ pyenv install 3.8.3
$ git clone https://github.com/cdberkstresser/fah-control.git 
$ cd fah-control
$ pyenv local 3.8.3
$ python -m pip install stdeb setuptools
$ python setup.py --command-packages=stdeb.command sdist_dsc
```

```
$python setup.py --command-packages=stdeb.command sdist_dsc
dir = .
running sdist_dsc
running egg_info
writing FAHControl.egg-info/PKG-INFO
writing dependency_links to FAHControl.egg-info/dependency_links.txt
writing requirements to FAHControl.egg-info/requires.txt
writing top-level names to FAHControl.egg-info/top_level.txt
reading manifest file 'FAHControl.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'FAHControl.egg-info/SOURCES.txt'
running sdist
running check
creating FAHControl-0.0.0
creating FAHControl-0.0.0/FAHControl.egg-info
creating FAHControl-0.0.0/fah
creating FAHControl-0.0.0/fah/db
creating FAHControl-0.0.0/fah/util
creating FAHControl-0.0.0/images
copying files to FAHControl-0.0.0...
copying FAHControl -> FAHControl-0.0.0
copying FAHControl.desktop -> FAHControl-0.0.0
copying MANIFEST.in -> FAHControl-0.0.0
copying README.md -> FAHControl-0.0.0
copying setup.py -> FAHControl-0.0.0
copying FAHControl.egg-info/PKG-INFO -> FAHControl-0.0.0/FAHControl.egg-info
copying FAHControl.egg-info/SOURCES.txt -> FAHControl-0.0.0/FAHControl.egg-info
copying FAHControl.egg-info/dependency_links.txt -> FAHControl-0.0.0/FAHControl.egg-info
copying FAHControl.egg-info/requires.txt -> FAHControl-0.0.0/FAHControl.egg-info
copying FAHControl.egg-info/top_level.txt -> FAHControl-0.0.0/FAHControl.egg-info
copying fah/Client.py -> FAHControl-0.0.0/fah
copying fah/ClientConfig.py -> FAHControl-0.0.0/fah
copying fah/Connection.py -> FAHControl-0.0.0/fah
copying fah/FAHControl.py -> FAHControl-0.0.0/fah
copying fah/FAHControl_glade.py -> FAHControl-0.0.0/fah
copying fah/SlotConfig.py -> FAHControl-0.0.0/fah
copying fah/Version.py -> FAHControl-0.0.0/fah
copying fah/WidgetMap.py -> FAHControl-0.0.0/fah
copying fah/__init__.py -> FAHControl-0.0.0/fah
copying fah/wraplabel.py -> FAHControl-0.0.0/fah
copying fah/db/Column.py -> FAHControl-0.0.0/fah/db
copying fah/db/Database.py -> FAHControl-0.0.0/fah/db
copying fah/db/Table.py -> FAHControl-0.0.0/fah/db
copying fah/db/__init__.py -> FAHControl-0.0.0/fah/db
copying fah/util/EntryValidator.py -> FAHControl-0.0.0/fah/util
copying fah/util/OrderedDict.py -> FAHControl-0.0.0/fah/util
copying fah/util/PasswordValidator.py -> FAHControl-0.0.0/fah/util
copying fah/util/SingleApp.py -> FAHControl-0.0.0/fah/util
copying fah/util/__init__.py -> FAHControl-0.0.0/fah/util
copying images/FAHControl.ico -> FAHControl-0.0.0/images
copying images/FAHControl.png -> FAHControl-0.0.0/images
Writing FAHControl-0.0.0/setup.cfg
Creating tar archive
removing 'FAHControl-0.0.0' (and everything under it)
This version of stdeb requires python-all >= 2.6.6-3, but you do not have this package installed. Could not check compatibility.
dpkg-buildpackage: info: source package fahcontrol
dpkg-buildpackage: info: source version 0.0.0-1
dpkg-buildpackage: info: source distribution unstable
dpkg-buildpackage: info: source changed by Joseph Coffland <joseph@cauldrondevelopment.com>
 dpkg-source --before-build .
dpkg-source: info: using options from fahcontrol-0.0.0/debian/source/options: --extend-diff-ignore=\.egg-info$
 fakeroot debian/rules clean
dh clean --with python3 --buildsystem=python_distutils
dh: warning: Compatibility levels before 9 are deprecated (level 7 in use)
dh: error: unable to load addon python3: Can't locate Debian/Debhelper/Sequence/python3.pm in @INC (you may need to install the Debian::Debhelper::Sequence::python3 module) (@INC contains: /etc/perl /usr/local/lib/x86_64-linux-gnu/perl/5.30.0 /usr/local/share/perl/5.30.0 /usr/lib/x86_64-linux-gnu/perl5/5.30 /usr/share/perl5 /usr/lib/x86_64-linux-gnu/perl/5.30 /usr/share/perl/5.30 /usr/local/lib/site_perl /usr/lib/x86_64-linux-gnu/perl-base) at (eval 3) line 1.
BEGIN failed--compilation aborted at (eval 3) line 1.

make: *** [debian/rules:7: clean] エラー 255
dpkg-buildpackage: error: fakeroot debian/rules clean subprocess returned exit status 2
Traceback (most recent call last):
  File "setup.py", line 119, in <module>
    setup(
  File "/home/yumetodo/.pyenv/versions/3.8.3/lib/python3.8/site-packages/setuptools/__init__.py", line 145, in setup
    return distutils.core.setup(**attrs)
  File "/home/yumetodo/.pyenv/versions/3.8.3/lib/python3.8/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/home/yumetodo/.pyenv/versions/3.8.3/lib/python3.8/distutils/dist.py", line 966, in run_commands
    self.run_command(cmd)
  File "/home/yumetodo/.pyenv/versions/3.8.3/lib/python3.8/distutils/dist.py", line 985, in run_command
    cmd_obj.run()
  File "/home/yumetodo/.local/lib/python3.8/site-packages/stdeb/command/sdist_dsc.py", line 135, in run
    build_dsc(debinfo,
  File "/home/yumetodo/.local/lib/python3.8/site-packages/stdeb/util.py", line 1441, in build_dsc
    dpkg_buildpackage('-S','-sa',*args,cwd=fullpath_repackaged_dirname)
  File "/home/yumetodo/.local/lib/python3.8/site-packages/stdeb/util.py", line 538, in dpkg_buildpackage
    process_command(args, cwd=cwd)
  File "/home/yumetodo/.local/lib/python3.8/site-packages/stdeb/util.py", line 191, in process_command
    check_call(args, cwd=cwd)
  File "/home/yumetodo/.local/lib/python3.8/site-packages/stdeb/util.py", line 46, in check_call
    raise CalledProcessError(retcode)
stdeb.util.CalledProcessError: 2
```

ffissore commented:
@cdberkstresser are the steps in the github action specific to ubuntu18 or do they work on your ubuntu20 as well? I haven't yet upgraded and github actions support only ubuntu18 at the moment

cdberkstresser commented:
@yumetodo @ffissore Unfortunately, I had differences between 18.04 in the github docker image, 18.04 desktop, and 20.04 desktop.  
On 20.04, I get a crash on the sdist or bdist commands and ended up using
`debuild -us -uc` to get an image on 20.04.

gardenappl commented:
For users of Arch Linux: **I made an AUR package with this fork here: https://aur.archlinux.org/packages/fahcontrol-gtk3-git/**

I made it mostly as a test, but it turns out that it works quite well.

*The only issue is icons, for some reason this fork uses relative paths for .ico files* (I assume this is temporary?). This breaks the usual setup.py procedure, as the image files get put in a different location. [https://aur.archlinux.org/cgit/aur.git/tree/fahcontrol-icon-path-linux.patch?h=fahcontrol-gtk3-git](I wrote a quick and dirty patch) to fix this on Arch, but as you can see, that patch isn't exactly cross-platform. I've never worked with Python or GTK before, so I couldn't come up with anything better.

Anyway, Arch users can install that package and this should work just fine.

ffissore commented:
@gardenappl any chance the .ico file can be left where it is rather than moved to `/usr/share/pixmaps` ?

ffissore commented:
ok, sorry my fault: it's setup that is "installing" that file under `/usr/share/pixmaps`: I'll fix it asap

ffissore committed:
Fixed wrong usage os "is" and "!="

Signed-off-by: Federico Fissore <federico@fissore.org>

ffissore committed:
Fixed wrong usage of mutable default arguments

Signed-off-by: Federico Fissore <federico@fissore.org>

ffissore committed:
Added requirements.txt

Signed-off-by: Federico Fissore <federico@fissore.org>

ffissore committed:
Look for icon in local path and under /usr/share/pixmaps

Signed-off-by: Federico Fissore <federico@fissore.org>

ffissore commented:
@gardenappl I've pushed a fix to address your feedback. Can you give it a try? Your patch could be removed now

gardenappl commented:
@ffissore Just checked it, seems to work fine without the patch now.

But I was kinda hoping that GTK3 would offer a more robust way of finding the icon path.

I've been doing a bit of searching and, apparently, the freedesktop.org standard [requires programs to look for icons in these folders](https://specifications.freedesktop.org/icon-theme-spec/icon-theme-spec-latest.html#directory_layout):

1.  $HOME/.icons
2.  $XDG_DATA_DIRS/icons
3. /usr/share/pixmaps

In that order.

ffissore committed:
Look for icon in local path and in other system folders as specified by https://specifications.freedesktop.org/icon-theme-spec/icon-theme-spec-latest.html#directory_layout

ffissore commented:
@gardenappl Done. Can you give it a try?

gardenappl commented:
@ffissore That worked fine. It kinda sucks that you had to hardcore those paths, though.

jcoffland committed:
Set client password from FAHControl, Fix for Linux dependencies

jcoffland committed:
Removed gpu-index, opencl-index and cuda-index config.

BryanQuigley commented:
Tested on Ubuntu 20.10 and it works for me.rked fine. It kinda sucks that you had to hardcode those paths, though.

ffissore committed:
Merge remote-tracking branch 'upstream/master' into master

ffissore commented:
FYI resolved conflicts and merged commits 3189958e4ea40516dc2956a4d6b1797ff75acd02 and ec5852b1766d9dba65c615dca7e59d1dcdc50fcd into this branch

Artoria2e5 commented:
Moving to Gtk3 would also help with HiDPI on Windows!

Davilink commented:
any progress ?

cdberkstresser commented:
@Davilink I'm not sure there is a viable path for this pull request to be accepted.  There seem to be substantial problems building a python3 executable on Windows.

Artoria2e5 commented:
Sounds fishy, will clone and try on Windows.

Artoria2e5 committed:
Merge https://github.com/FoldingAtHome/fah-control into master

Artoria2e5 committed:
PyON: properly attribute code source

This is sloppy.

Artoria2e5 committed:
Replace gtk module with gi for win32

Seems to be the only thing blocking build?

Artoria2e5 committed:
Tell cx_Freeze to bring the dlls

Still stuck on a from . import _gi
ModuleNotFoundError: No module named 'gi._error'

Artoria2e5 committed:
setup.py: include all of gi

This fixes all the Python-side issues. Now we are at Gtk scratching
its head over what a PNG is due to missing MIME.

ffissore commented:
Please do, @Artoria2e5: the windows ecosystem is full of users, and without hackers. We need a windows hacker to make this work

Artoria2e5 commented:
See https://github.com/cdberkstresser/fah-control/pull/19 for the Windows PR on this PR. Gtk is loading, but I have to feed it the MIME database too.

Artoria2e5 committed:
It works! Eureka!

Artoria2e5 committed:
Update README and SConstruct to py3

cdberkstresser committed:
Merge pull request #19 from Artoria2e5/master

A merge with upstream, and possibly windows stuff

Artoria2e5 commented:
Do we need a check on macOS too? I think I can convince a hackintosh to do that.

kblaschke committed:
Fix PyON JSON parsing and exception string formatting.

cdberkstresser committed:
Merge pull request #20 from kblaschke/master

Fix PyON JSON parsing and exception string formatting.

gardenappl commented:
Sorry if this is a silly question, but will this be [portable to GTK 4](https://developer.gnome.org/gtk4/unstable/gtk-migrating-3-to-4.html#id-1.7.4.3)?👀

gardenappl committed:
Update dependencies in README, add python-six

samy-k97 commented:
Hi, I just wanted to leave some instructions for anyone who wants to test this and isn't a programmer but can copy paste to the terminal.

Keep in mind I have very very basic knowledge of programming so correct me If something isn't right or missing

> git clone https://github.com/cdberkstresser/fah-control.git
>  cd fah-control
>  python3 setup.py build
>  ./FAHControl

cdberkstresser committed:
Merge pull request #21 from gardenappl/patch-1

Update dependencies in README, add python-six

yumetodo commented:
On Windows 10 msys2 mingw64 

```bash
$python -V
Python 3.8.7
$git clone https://github.com/FoldingAtHome/fah-control.git
$cd fah-control/
$git remote add cdberkstresser https://github.com/cdberkstresser/fah-control.git
$git fetch --all
$git switch -c pr70 cdberkstresser/master
$python3 setup.py build
dir = .
('C:/msys64/mingw64/lib/gdk-pixbuf-2.0/2.10.0/loaders.cache', '/lib/gdk-pixbuf-2.0/2.10.0/loaders.cache')
('C:/msys64/mingw64/lib/gdk-pixbuf-2.0/2.10.0/loaders/io-wmf.dll', '/lib/gdk-pixbuf-2.0/2.10.0/loaders/io-wmf.dll')
('C:/msys64/mingw64/lib/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-ani.dll', '/lib/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-ani.dll')
('C:/msys64/mingw64/lib/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-bmp.dll', '/lib/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-bmp.dll')
('C:/msys64/mingw64/lib/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-gif.dll', '/lib/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-gif.dll')
('C:/msys64/mingw64/lib/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-icns.dll', '/lib/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-icns.dll')
('C:/msys64/mingw64/lib/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-ico.dll', '/lib/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-ico.dll')
('C:/msys64/mingw64/lib/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-jpeg.dll', '/lib/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-jpeg.dll')
('C:/msys64/mingw64/lib/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-png.dll', '/lib/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-png.dll')
('C:/msys64/mingw64/lib/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-pnm.dll', '/lib/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-pnm.dll')
('C:/msys64/mingw64/lib/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-qtif.dll', '/lib/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-qtif.dll')
('C:/msys64/mingw64/lib/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-svg.dll', '/lib/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-svg.dll')
('C:/msys64/mingw64/lib/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-tga.dll', '/lib/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-tga.dll')
('C:/msys64/mingw64/lib/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-tiff.dll', '/lib/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-tiff.dll')
('C:/msys64/mingw64/lib/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-xbm.dll', '/lib/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-xbm.dll')
('C:/msys64/mingw64/lib/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-xpm.dll', '/lib/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-xpm.dll')
running build
running build_exe
Traceback (most recent call last):
  File "setup.py", line 186, in <module>
    setup(
  File "C:/msys64/mingw64/lib/python3.8/site-packages/cx_Freeze/dist.py", line 392, in setup
    distutils.core.setup(**attrs)
  File "C:/msys64/mingw64/lib/python3.8/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "C:/msys64/mingw64/lib/python3.8/distutils/dist.py", line 966, in run_commands
    self.run_command(cmd)
  File "C:/msys64/mingw64/lib/python3.8/distutils/dist.py", line 985, in run_command
    cmd_obj.run()
  File "C:/msys64/mingw64/lib/python3.8/distutils/command/build.py", line 135, in run
    self.run_command(cmd_name)
  File "C:/msys64/mingw64/lib/python3.8/distutils/cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "C:/msys64/mingw64/lib/python3.8/distutils/dist.py", line 985, in run_command
    cmd_obj.run()
  File "C:/msys64/mingw64/lib/python3.8/site-packages/cx_Freeze/dist.py", line 234, in run
    freezer = cx_Freeze.Freezer(
  File "C:/msys64/mingw64/lib/python3.8/site-packages/cx_Freeze/freezer.py", line 85, in __init__
    self.includeFiles = process_path_specs(includeFiles)
  File "C:/msys64/mingw64/lib/python3.8/site-packages/cx_Freeze/common.py", line 74, in process_path_specs
    raise ConfigError(
cx_Freeze.common.ConfigError: target path for include file may not be an absolute path
```

Artoria2e5 commented:
It doesn't seem to affect much does it? The thing isn't present in the ls listing after all. Something else must have deleted it before the other deleter did.

chabgood commented:
> Hi, I just wanted to leave some instructions for anyone who wants to test this and isn't a programmer but can copy paste to the terminal.
> 
> Keep in mind I have very very basic knowledge of programming so correct me If something isn't right or missing
> 
> > git clone https://github.com/cdberkstresser/fah-control.git
> > cd fah-control
> > python3 setup.py build
> > ./FAHControl

on Ubuntu, `sudo apt-get install -y python3-setuptools`

FalconFour commented:
> Hi, I just wanted to leave some instructions for anyone who wants to test this and isn't a programmer but can copy paste to the terminal.
> 
> Keep in mind I have very very basic knowledge of programming so correct me If something isn't right or missing
> 
> > git clone https://github.com/cdberkstresser/fah-control.git
> > cd fah-control
> > python3 setup.py build
> > ./FAHControl

Pretty darn cool and probably 95% of the headache fixed, but I can't get past it. I'm trying to build on Windows so I can get a FAHControl interface on a PC I don't want to install F@H on. Plus, I would really like to see the progress of updating the FAHControl interface, so I thought I'd pull this and try running it. But alas, no love - still no love, been bashing my head against these cryptic errors for about 10 minutes now.

```
dir = C:\Users\***\Desktop\fah-control
Traceback (most recent call last):
  File "C:\Users\***\Desktop\fah-control\setup.py", line 73, in <module>
    from cx_Freeze import setup, Executable
ModuleNotFoundError: No module named 'cx_Freeze'
```

I can't get around this "cx_Freeze" issue, and Googling that error just gives me developer resources but no hint of how to install this "cx_freeze" thing.

Artoria2e5 commented:
@FalconFour I recommend you do it using MSYS2, specifically its MinGW sub-environment. Check out the install command in https://github.com/cdberkstresser/fah-control.

dchalmers commented:
> > Hi, I just wanted to leave some instructions for anyone who wants to test this and isn't a programmer but can copy paste to the terminal.
> > Keep in mind I have very very basic knowledge of programming so correct me If something isn't right or missing
> > > git clone https://github.com/cdberkstresser/fah-control.git
> > > cd fah-control
> > > python3 setup.py build
> > > ./FAHControl
> 
> on Ubuntu, `sudo apt-get install -y python3-setuptools`

The above may fail if apt is still complaining of unmet dependencies (from the attempt to install the Python2 version).
`sudo apt --fix-broken install`
`sudo apt update && sudo apt upgrade`
should fix, and enable python3-setuptools to be installed.

mlmatlock committed:
Change module to import 'MutableMapping' variable

cornytrace commented:
This program is currently the only package left on my system still needing python2, it would be great if this could be merged.

seoriy commented:
@jcoffland, sorry for bothering you, but is there any updates on this PR? My Nvidia RTX 3060 is waiting to run some tasks from F@H, but instead of that, I'm wasting my time trying to fix dependencies issues on Ubuntu 21.10, because fahcontrol is still depend on Python2 :( I'm sure, someone other is also will be happy to see this PR or similar done...

immibis commented:
Yes, this should be treated as the program being totally broken. Python2 is totally out of support and it's a bad idea to have it installed on your system even if you're able to.

immibis commented:
The good news is you can clone cdberkstresser's branch and install it with

    python3 setup.py build
    python3 setup.py install

and it works on Python 3

watmm commented:
Hi all.

This may be unrelated to the Python 3 version above but maybe someone can help.

I just installed this version (via pip would be better) and then deleted the suggested slots in an attempt to get it to only use my GPU. Now i'm wondering how to restore it to the originally suggested slots and slot options.

Reinstalling doesn't help. Any ideas? 

Cheers.

cdberkstresser committed:
Merge pull request #23 from mlmatlock/python_3.10_fix

Change module to import 'MutableMapping' variable

nkorell1970 commented:
I would like to commend all who have worked on this effort to fix what so desperately needed doing.  I am a self proclaimed newbie who is trying to continue to run a Ubuntu box for the sake of FaH.  Needless to say I ran into the Python dependcy wall and have been dead in the water for months.  What is the correct sequence of how to install the FaH software and correct the dependencies. Could someone please post a quick set of instructions or point me in the direction of somewhat may be already complete?  I would like to  correctly utilize this code.  I have a fresh install of 20.04.

TheFlipside commented:
I would carefully like to ask about the status of this, there are multiple PR which all aim for the transition to Python3 but the overall process seems to be stalled.

okaestne committed:
debian: use debhelper 11; update depends

needed to be able to build .deb on ubuntu22

okaestne committed:
debian: ignore temp files and windows dlls

okaestne committed:
update .gitignore

okaestne committed:
debian/rules: add newline at end of file

okaestne committed:
debian: install .desktop file (fixes #17)

cdberkstresser committed:
Merge pull request #27 from okaestne/fix-dpkg

Improvements to debian packaging

cdberkstresser committed:
compatibility updates

cdberkstresser committed:
resolving setup.py conflicts

cdberkstresser committed:
Adding restore of size / position of main window

cdberkstresser committed:
removing window size options from prefs since auto saves

Artoria2e5 commented:
> I would carefully like to ask about the status of this, there are multiple PR which all aim for the transition to Python3 but the overall process seems to be stalled.

This is *the* PR to work on, because it also does Gtk3 transition. Last time I checked it works fine on Windows. You can PR the PR at https://github.com/cdberkstresser/fah-control/pulls.

I'm actually pretty surprised to see it in my notification again!

cdberkstresser commented:
@Artoria2e5 I don't work for FaH, so just my personal guess here.  I'm expecting FAHControl itself to obsolete and die along with this PR.  Their [beta](https://foldingathome.org/beta) is moving to a more web ui direction bundled with the client.  I just added a bit to this PR recently to keep FAHControl usable until that happens.

TheFlipside commented:
if FAHControl would become obsolete and die as you describe I won't let that happen and would personally fork it and continue to make it work

FalconFour commented:
Going back a layer, I guess the question is - instead of kicking along this old software along, what use cases does it serve that the web interface (which seems to be the primary point of development) doesn't? I would imagine that the web UI dev team would be well served to listen closely to the use cases of people still needing FAHControl in order to integrate those features and use cases into the Web UI.

For me, it's when doing more advanced setup of dedicated systems, and controlling/managing multiple systems in a dashboard - maybe you could call it an "enterprise interface" for larger-scale dedicated FAH systems. The web interface feels like Little Tikes / Fisher-Price compared to the controls available in FAHControl, and that's why I immediately close the web interface and open up FAHControl whenever I work with FAH. There's a LOT you can't see or configure in the web interface, especially when working with weird high-performance setups.

If there were better controls available in the web interface, I would imagine none of us would even be trying to shoehorn Python2 into a modern system to get FAHControl running.

If there is no such effort underway already to bring more advanced configuration and diagnostics to the web UI, then yeah, definitely, there's a need to port FAHControl into the current decade. Just wondering where the team's priorities are.

TheFlipside commented:
Ah yes I see, well I don't wanna be an obstacle in the way of modern progress but I do wish for an application which offers the same functionality as FAHControl today; mostly to be able to monitor and control multiple systems in one UI, I don't want to have to open multiple browser tabs to get access to all my systems.
And as you said it should also offer the possibilities to advanced settings like currently possible with FAHControl.

cdberkstresser committed:
Update packagedeb.yml to ubuntu 22.04

cdberkstresser committed:
Update packagedeb.yml to require dh-python

cdberkstresser committed:
Update packagedeb.yml python v4

cdberkstresser committed:
Update packagedeb.yml python 3.11.3

cdberkstresser committed:
Update packagedeb.yml again

cdberkstresser committed:
Update packagedeb.yml who cares

cdberkstresser committed:
Update packagedeb.yml libgirepository1.0-dev

cdberkstresser committed:
Update packagedeb.yml pygobject

Wedge009 commented:
> ...instead of kicking along this old software along, what use cases does it serve that the web interface (which seems to be the primary point of development) doesn't?

I never really used the web interface as I found it unreliable/unresponsive. Also, I mainly use the 'advanced' interface because I can monitor multiple hosts from my main computer - I haven't seen if the web interface can do that.

I admit I only discovered this repository today - since the most recent stable release was in October 2020 (that's nearly three years ago at time of writing), I thought there wasn't any active development on the FAH platform. I haven't read through all the comments here but are there major reasons for this not yet being merged (and a new release made available)? I saw the comment that this branch can be run manually against Python 3, but for Ubuntu 22.04 I've just been manually installing old Python 2 packages from older Ubuntu releases to continue using FAH on Linux. I suppose that may not be feasible by the time 24.04 comes along (and that's less than a year away).

jcoffland commented:
Current development efforts are focused on the new v8 client which is in beta.  https://github.com/FoldingAtHome/fah-client-bastet & https://github.com/FoldingAtHome/fah-web-client-bastet

Wedge009 commented:
Thanks for the pointer.
